libsolidity/interface/CompilerStack.cpp

```diff
@@ -1212,30 +1218,31 @@ void CompilerStack::compileContract(
 	_otherCompilers[compiledContract.contract] = compiler;
 }
 
-void CompilerStack::generateIR(ContractDefinition const& _contract)
+void CompilerStack::generateIR(
+	ContractDefinition const& _contract,
+	map<ContractDefinition const*, string const>& _otherYulSources
+)
 {
 	solAssert(m_stackState >= AnalysisPerformed, "");
 	if (m_hasError)
 		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Called generateIR with errors."));
 
-	if (!_contract.canBeDeployed())
-		return;
-
-	map<ContractDefinition const*, string const> otherYulSources;
-
 	Contract& compiledContract = m_contracts.at(_contract.fullyQualifiedName());
 	if (!compiledContract.yulIR.empty())
 		return;
 
 	string dependenciesSource;
 	for (auto const* dependency: _contract.annotation().contractDependencies)
 	{
-		generateIR(*dependency);
-		otherYulSources.emplace(dependency, m_contracts.at(dependency->fullyQualifiedName()).yulIR);
+		generateIR(*dependency, _otherYulSources);
+		_otherYulSources.emplace(dependency, m_contracts.at(dependency->fullyQualifiedName()).yulIR);
 	}
 
+	if (!_contract.canBeDeployed())
```

Comment:
If you want to merge this quickly, can you create a PR that only has this change here? I would really try to avoid adding the other complication.

Reply:
Well, doing this alone doesn't really fixes anything..

Reply:
However, I pushed a version here now that only fixes the non-yul parts.